python/cuda_parallel/cuda/parallel/experimental/_iterators.py

```diff
@@ -100,13 +103,13 @@ def __init__(self, ptr, ntype):
             _ncc(
                 f"{self.prefix}_advance",
                 RawPointer.pointer_advance,
-                numba.types.void(data_as_ntype_pp, _DISTANCE_NUMBA_TYPE),
-            ),
+                (data_as_ntype_pp, _DISTANCE_NUMBA_TYPE),
```

Comment:
This one I'd be tempted to keep:
* It's more readable, because the entire signature is immediately obvious.
* We're sure we want `void` here.
* I'm guessing it's less work for numba, because I'd expect that it doesn't run the code path for inferring the return type.

But I'm fine either way.

Reply:
I would second that. There is a benefit to being consistent and one of the main benefits is that these places where we intentionally want something else stand out more

Reply:
I changed all the signatures of the `*advance` methods to include an explicit `void` return type